pycbc/events/single.py

```diff
@@ -31,31 +54,65 @@ def from_cli(cls, args, ifo):
         return cls(
            ifo, newsnr_threshold=args.single_newsnr_threshold[ifo],
            reduced_chisq_threshold=args.single_reduced_chisq_threshold[ifo],
-           fixed_ifar=args.single_fixed_ifar[ifo],
            duration_threshold=args.single_duration_threshold[ifo],
+           fit_file=h5py.File(args.fit_definition_file),
```

Comment:
Could you instead open the file with a `with` statement, read whatever data is needed, and pass it to the `cls` constructor? I would like to minimize the number of open files during operation.

Reply:
Also, you probably want to open the file explicitly in read-only mode.

Reply:
Yeah I had wanted to avoid having it open for too long, but there was an issue with having too many attributes for the class - I got around that by using dictionaries as attributes